### ISSUE TITLE
Warning on the Cart Rules creation/editing page

### ISSUE THREAD

**Issue by Codencode** [PrestaShop/PrestaShop#32654]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

When accessing the cart rule creation/modification page, the following warning appears:
```
Warning on line 81 in file XXX\classes\Smarty\SmartyLazyRegister.php
[2] end(): Argument #1 ($array) must be passed by reference, value given
```

I attach a video.

https://github.com/PrestaShop/PrestaShop/assets/2463965/cbabeec0-3161-44de-a580-59e845831d77

### Steps to reproduce

1. In admin go to Catalog > Discounts
2. Click on **Add new cart rule**
3. The warning is displayed

### PrestaShop version(s) where the bug happened

8.0.4, 8.1.x

### PHP version(s) where the bug happened

all

**Comment by hibatallahAouadni** [comment]
Hello @Codencode 

I didn't manage to reproduce the issue with PS804 and PS81x (PHP7.4), see the attached screen record below:

[issue_32654.webm](https://github.com/PrestaShop/PrestaShop/assets/23308427/028da007-7401-4911-ae23-9e3ddad38532)

Please check and feedback.

Thanks!

**Comment by Codencode** [comment]
Hi @hibatallahAouadni,

I don't know how that's possible, but it happens to me.
This problem is similar to the one that occurs in this issue: #32680, can you check this problem too? 

Thank you!

**Comment by RosaBenouamer** [comment]
@hibatallahAouadni : it seems to be the same message bu not on the same page so perhaps conserve the two issues ? It's nevertheless interesting to correlate the two :-) if you are ok I put it in the kanban at the same time as [32680](https://github.com/PrestaShop/PrestaShop/issues/32680). Can we take off Waiting for author label ?

**Comment by hibatallahAouadni** [comment]
Hello @RosaBenouamer 

I didn't reproduce the issue with the provided infos :disappointed: 
So, @Codencode could you please provide more infos to help me reprodcue the issue :pray: 

Thanks!

**Comment by Codencode** [comment]
Hi @hibatallahAouadni,

you must activate the "debug" mode which allows you to view the error, in the video with the 8.0.4 version you have the "debug" mode deactivated, while in the 8.1.0 version you have active debugging, so you should see the error .
In my opinion, the version you downloaded of 8.1.x already includes the fix for this issue, for which I have already created a PR that has already been merged with 8.1.x.
